src/cli/onefuzz/api.py

```diff
@@ -970,6 +970,58 @@ def list(
             results[container] = self.onefuzz.containers.files.list(container).files
         return results
 
+    def delete(self, job_id: UUID_EXPANSION, *, only_job_specific: bool = True) -> None:
+        SAFE_TO_REMOVE = [
+            enums.ContainerType.crashes,
+            enums.ContainerType.setup,
+            enums.ContainerType.inputs,
+            enums.ContainerType.reports,
+            enums.ContainerType.unique_inputs,
+            enums.ContainerType.unique_reports,
+            enums.ContainerType.no_repro,
+            enums.ContainerType.analysis,
+            enums.ContainerType.coverage,
+            enums.ContainerType.readonly_inputs,
+            enums.ContainerType.regression_reports,
+        ]
+
+        job = self.onefuzz.jobs.get(job_id)
+        to_delete = set()
+        to_keep = set()
+        for task in self.onefuzz.jobs.tasks.list(job_id=job.job_id):
+            for container in task.config.containers:
+                if container.type not in SAFE_TO_REMOVE:
+                    to_keep.add(container.name)
+                elif not only_job_specific:
+                    to_delete.add(container.name)
+                elif only_job_specific and (
+                    self.onefuzz.utils.build_container_name(
+                        container_type=container.type,
+                        project=job.config.project,
+                        name=job.config.name,
+                        build=job.config.build,
+                        platform=task.os,
+                    )
+                    == container.name
+                ):
+                    to_delete.add(container.name)
+                else:
+                    to_keep.add(container.name)
+
+        for container_name in to_keep:
```

Comment:
In the absence of unit testing, it'd be nice if we were using non-intersecting sets / a structure that partitioned the containers.

I don't think you need to change it in this PR (your comment below helps a lot), but since this is about deletion, I'm inclined to be extra cautious.

Reply:
added